### ISSUE TITLE
OAuth verifier fails signature with header_auth=True

### ISSUE THREAD

**Issue by samuraisam** [maraujop/requests-oauth#16]
When fetching the access token using header auth, the signature base is incorrectly computed.

Doesn't work:

```
# create an oauth hook with the consumer and request token values
hook = OAuthHook(..., header_auth=True)
hook.token.verifier = get_verifier_from_redirect(previous_request)
sesh = requests.session(hooks={'pre_request': hook})
sesh.get('http://something.com/oauth/access_token').content == 'Invalid signing base'
```

Does work:

```
# create an oauth hook with the consumer and request token values
hook = OAuthHook(...)
sesh = requests.session(hooks={'pre_request': hook})
sesh.get('http://something.com/oauth/access_token', 
          params={'oauth_verifier': get_verifier_from_redirect(previous_request)})
```

**Comment by maraujop** [comment]
Yes, you are right Samuel, thanks for reporting it.

I'm working on fixing this one. There is a bug in requests itself, that is blocking part of a fix for this problem.
https://github.com/kennethreitz/requests/issues/445

I'm gonna start with a partial fix asap and will get the rest fixed when my fix for this issue gets merged.

Cheers,
Miguel

**Comment by maraujop** [comment]
Version 0.4.0 has been released, it works well with requests 0.12.1 or higher. I'm closing this issue.

Cheers
